documentation/add/07-sqw_redesign.md

```diff
@@ -1,58 +1,84 @@
+The structure of generic `sqw` object is presented on Fig.2:
 
-![SQW Class Overview](../diagrams/sqw.png)
+![Fig.2. SQW Class Overview](../diagrams/sqw.png)
 
 ### DnDBase, DnD
 
-The `DnD` object is a "cut-down SQW" object containing only `Image` data. This exists in n-dimensional forms, with each class extending an abstract base class. The `PixelData`, `IX_Instr`and `IX_DetectorArray` information is NOT included, and any data manipulation operations are performed directly on the `Image` data.
+The `DnD` object exist as part of  `sqw` object but also can exist as "cut-down `sqw`" object containing only multidimensional `Image` data expressed in the coordinate system of interest. The "Image" exists in 0 to 4 dimensional forms, with each particular number of dimensions class defining the abstract base class. The `PixelData`, `main_header` and `Experiment` information is NOT included, and any data manipulation operations are performed directly on the `Image` data.
+
+Extraction from the whole inheritance diagram for the `DnD` objects is presented on the Fig.3:
+
+![Fig.3. DnD Class inheritance diagram](../diagrams/DND-inheritance.png)
+
+The diagram also shows `data_sqw_dnd` object inheriting from `DnDBase` and containing arbitrary number of dimensions unlike other `DnD` objects which define their specific number of dimensions.  This object left for IO compatibility with previous versions of Horace code and is not used in Horace for any other purpose except IO restoring old data.
+
+The `DnDBase` base class is an abstract class holding the common data and common code, including the operation manager which is responsible for matching dimensions between the specific `DnD` objects before executing appropriate algorithms. The structure of `DnDBase` class together with its children, defining specific number of dimensions is presented on Fig.4:
+
+![Fig.4. DND Class Overview](../diagrams/dnd.png)
+
+The diagram shows that any `DnD` object contains *signal*, *error* and *npix* (*npix* describes the number of pixels (neutron events) contributed to the appropriate area of the image and related to the image scaling) arrays of appropriate dimensions and two additional properties, namely *proj* and *axes*. *Proj* field contains the instance of `aProjection` class, which defines the transformation from the coordinate system of `PixelData` class to the `Image` (`DnD` object) coordinate system. For example, in the most common case of `Image` coordinate system being `hkl-dE` coordinate system and `PixelData` class coordinate system being Crystal Cartesian coordinate system the transformation is the matrix multiplication of the `PixelData` coordinates by `UB` matrix of Bussing and Levy [^1].
```

Comment:
'area' of the image - I would prefer image pixel, or voxel, but making it clear that this is the smallest area of the image rather than an extended region
put contributed to the appropriate [voxel] of the image before (*npix. The function of 'related to the image scaling' is unclear and the bracket after scaling is not paired.

Reply:
I would leave `area` as it is not only voxel but any chosen sequence of voxels, Otherwise, hopefully reformatted appropriately.